common/path.go

```diff
@@ -100,14 +100,28 @@ func DefaultAssetPath() string {
 }
 
 func DefaultDataDir() string {
-	usr, _ := user.Current()
-	if runtime.GOOS == "darwin" {
-		return filepath.Join(usr.HomeDir, "Library", "Ethereum")
-	} else if runtime.GOOS == "windows" {
-		return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
-	} else {
-		return filepath.Join(usr.HomeDir, ".ethereum")
+	// Try to place the data folder in the user's home dir
+	if usr, err := user.Current(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(usr.HomeDir, "Library", "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
+		} else {
+			return filepath.Join(usr.HomeDir, ".ethereum")
+		}
+	}
+	// If the user cannot be retrieved (Android, Fedora/Go bug), use the current folder
+	if dir, err := os.Getwd(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(dir, "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(dir, "Ethereum")
+		} else {
```

Comment:
You could try using `os.Getenv("HOME")` as a fallback. 
That should also work on Fedora in most cases.